refactor(app): add and use deck map component in interventionmodal

sfoster1 opened the pull request:
Adds DeckMapContent to intervention modal. This is a small wrapper around one of two different kinds of deck map:
- A labware-rendering map that is also capable of rendering highlights around some labware, used in some selection screens in error recovery
- A deck config style block map that supports clicking on some of the blocks to select them, used in some selection screens in the drop tip wizard (and thus in error recovery)

Stories for the deckmap are [on storybook](https://s3-us-west-2.amazonaws.com/opentrons-components/exec-501-deckmap-content/index.html?path=/docs/app-molecules-interventionmodal-deckmapcontent--docs), where it's rendered side-by-side with the standin since it is only destined to be used in a two-column layout.

Extra fun changes:
- Use the ODD text size args for _all_ slot labels in `BaseDeck`, because otherwise they are completely unreadable once the deckmap gets to small. I think this looks a lot better everywhere, and is actually readable when the deckmap is small, so let's go with it; may want to come back and make these something more specific on desktop
- Error recovery has a `RecoveryMap`, a large data structure that is core to the wizard flow and defines how users move between steps and screens. Error recovery also had a `RecoveryMap`, which was a component for rendering a deckmap, and a `useRecoveryMapUtils`, a hook for getting that component's arguments. Now it uses the `DeckMapContent` component above directly, and `useDeckMapUtils`, a hook for getting that component's arguments
- There's something about the deck config style map that isn't quite right and I think that it's because of this: https://opentrons.atlassian.net/browse/EXEC-513

Review requests:
- i'm pretty sure what I did to drop tip will work but it might be reasonably called somewhat gross. this is all somewhat gross though, so c'est la vie
- the deckmap slot labels do not render right in firefox at these sizes. if you're looking at storybook, you're going to have to use chrome
- fix an issue that was preventing ER wizard from being displayed on ODD
Testing, todo:
- [ ] error recovery
- [ ] drop tip in error recovery
- [ ] drop tip outside of error recovery


Closes EXEC-501

sfoster1 committed:
and after all, why not? it's mine, isn't it?

my own? my... precious?

sfoster1 committed:
refactor: use intervention map in drop tip wizard

This should be a visually indistinguishable reimplementation.

sfoster1 committed:
refactor: use deckmap in error recovery

Two places in error recovery, refactoring as we go.

One big change was that there was a bunch of stuff called RecoveryMap as
in recovery (deck) map, but recovery also has the Recovery Map, the big
data structure that holds how all the steps of error recovery flow
together. This is very confusing, so now what used to be
RecoveryMap (component, utils) is now DeckMap.

sfoster1 committed:
fix up remaining tests

sfoster1 committed:
fix up some render issues

sfoster1 committed:
fix a logic error that prevented recovery from running

TamarZanzouri commented on app/src/molecules/InterventionModal/DeckMapContent.stories.tsx:
```diff
@@ -0,0 +1,157 @@
+  },
+]
+
+const CONSOLE_LOG_ON_SELECT = (location: ModuleLocation): void => {
```
nice!

TamarZanzouri commented on app/src/organisms/ErrorRecoveryFlows/RunPausedSplash.tsx:
```diff
@@ -44,7 +44,7 @@ export function useRunPausedSplash(
 ): boolean {
   // Don't show the splash when desktop ER wizard is active.
-  return !(!isOnDevice && showERWizard)
+  return isOnDevice && !showERWizard
```
Thank you!

TamarZanzouri commented on app/src/organisms/ErrorRecoveryFlows/hooks/useDeckMapUtils.ts:
```diff
@@ -21,29 +19,33 @@ import type {
+  modulesOnDeck: RunCurrentModulesOnDeck[]
+  labwareOnDeck: RunCurrentLabwareOnDeck[]
+  highlightLabwareEventuallyIn: string[]
+  kind: 'intervention'
```
just want to make sure we always want the kind to be intervention?

sfoster1 replied:
yeah. in this case we do because this is only picked up by error recovery use cases which always use that style of deck map. the deck config style is only used by the drop tip wiz

TamarZanzouri reviewed (approved):
code changes look really good! so much cleaner! I posted a question but besides that looks great

sfoster1 committed:
docs page: allow non-specified onSelect

sfoster1 committed:
refactor(app): separate StandIn story util

We're probably going to use this a lot more now that we'll have a bunch
of content components that live in TwoColumn, so let's make StandIn reusable.

sfoster1 committed:
feat(app): add DeckMapContent

add deckmap fixtures and such

more deckmap changes
